product_contract/models/product_product.py

```diff
@@ -0,0 +1,56 @@
+# Copyright 2020 ACSONE SA/NV.
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from odoo import _, api, fields, models
+from odoo.exceptions import ValidationError
+
+
+class ProductProduct(models.Model):
+    """Adds the possibility to override the template contract at variant level."""
+
+    _inherit = "product.product"
+
+    property_contract_template_id = fields.Many2one(
+        comodel_name="contract.template",
+        string="Contract Template",
+        company_dependent=True,
+    )
+
+    def _format_names(self, limit=None):
```

Comment:
```suggestion
    def _format_names_for_constraint_message(self, limit=None):
```
Just for code readability and to be confused with variant display name methods